src/Particle/Par_Init_ByFile.cpp

```diff
@@ -129,13 +133,30 @@ void Par_Init_ByFile()
 //    collect data for the target particle
 //    [id][att]
       if ( amr->Par->ParICFormat == PAR_IC_FORMAT_ID_ATT )
-         memcpy( ParData1, ParData_ThisRank+p*NParAtt, NParAtt*sizeof(real) );
+      {
+         if ( sizeof(real) == load_data_size )
+            memcpy( ParData1, ParData_ThisRank+p*NParAtt*load_data_size, NParAtt*load_data_size );
+         else
```

Comment:
It seems redundant. Can we remove it and rely on `if ( PAR_IC_FLOAT8 ) ... else ...`?

Reply:
@hyschive : I think if there is no merit for using `memcpy` to copy the whole consecutive memory than copying elementwise by `for` loop when filling `ParData1`, we can remove this `memcpy` line.

Reply:
Let's remove it